AWSAppSyncClient/Internal/AppSyncSubscriptionWithSync.swift

```diff
@@ -0,0 +1,596 @@
+            self.performSync()
+        }
+    }
+
+    // MARK: - Sync time
+
+    /// Fetches last sync time from local cache and updates `syncStrategy` with the retrieved value
+    private func initializeLastSyncTimeFromCache() {
+        do {
+            let lastSyncTime = try self.subscriptionMetadataCache?.getLastSyncTime(operationHash: getOperationHash())
+            syncStrategy.lastSyncTime = lastSyncTime
+            AppSyncLog.debug("Loaded lastSyncTime from cache: \(lastSyncTime.debugDescription)")
+        } catch {
+            // could not find it in cache, assume no sync was done previously
+        }
+    }
+
+    /// Updates the last sync time and persists it to `subscriptionMetadataCache` if present. Expected to
+    /// be called when subscription message is given to the caller, or if base query or delta query is run.
+    ///
+    /// - Parameters:
+    ///   - date: The date at which the last sync was successfully performed
+    private func setLastSyncTime(date: Date) {
+        do {
+            let adjustedDate = date.addingTimeInterval(TimeInterval.init(exactly: -2)!)
+            AppSyncLog.debug("Updating lastSync time \(adjustedDate)")
+            self.syncStrategy.lastSyncTime = adjustedDate
+            try subscriptionMetadataCache?.updateLastSyncTime(for: getOperationHash(), with: adjustedDate)
+        } catch {
+            // ignore cache write failure, be updated in next operation, is backed up by in-memory cache
+        }
+    }
+
+    // MARK: - Base query handling
+
+    /// Load base query results from cache, to quickly provide callers with the most recent results
+    private func initializeBaseQueryResultsFromCache() {
+        guard let appSyncClient = appSyncClient else {
+            return
+        }
+
+        let dispatchGroup = DispatchGroup()
+        dispatchGroup.enter()
+        AppSyncLog.info("Initializing base query results from cache")
+        appSyncClient.fetch(query: baseQuery, cachePolicy: CachePolicy.returnCacheDataDontFetch) { [weak self] (result, error) in
+            self?.baseQueryHandler(result, error)
+            dispatchGroup.leave()
+        }
+        dispatchGroup.wait()
+    }
+
+    /// Load base query results from service, forcing a bypass of the local cache
+    private func runBaseQueryFromService() -> Bool {
+        guard let appSyncClient = appSyncClient else {
+            return false
+        }
+
+        var success: Bool = true
+
+        AppSyncLog.info("Refreshing base query results from service")
+
+        let semaphore = DispatchSemaphore(value: 0)
+
+        let networkFetchTime = Date()
+
+        appSyncClient.fetch(query: baseQuery, cachePolicy: .fetchIgnoringCacheData) { [weak self] (result, error) in
+
+            defer {
+                semaphore.signal()
+            }
+
+            guard let self = self else {
+                return
+            }
+
+            // call customer if successful or error
```

Comment:
nitpick: `customer` -> `developer`

Reply:
Simplified the call and obviated the comment :)